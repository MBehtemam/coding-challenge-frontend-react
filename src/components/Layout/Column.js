import styled from 'styled-components';

const Column = styled.div`
  flex: ${props => (props.span ? props.span : 1)};
  justify-content: space-between;
  margin-left: 10px;
  margin-right: 10px;
`;
export default Column;
