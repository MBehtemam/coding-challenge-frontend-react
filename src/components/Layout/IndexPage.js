import React, { Fragment, Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import SearchBar from '../SearchBar';
import IncidentsGrid from '../IncidentsGrid';
import * as IncidentsActions from '../../logic/Actions/incidentsActions';

class IndexPage extends Component {
  componentDidMount() {
    const { getIncidents } = this.props;
    getIncidents();
  }

  render() {
    return (
      <Fragment>
        <SearchBar />
        <IncidentsGrid />
      </Fragment>
    );
  }
}

IndexPage.propTypes = {
  getIncidents: PropTypes.func
};
IndexPage.defaultProps = {
  getIncidents: IncidentsActions.getIncidents
};
const mapDispatchToProps = dispatch => ({
  getIncidents: () => {
    dispatch(IncidentsActions.getIncidents());
  }
});
export default connect(
  undefined,
  mapDispatchToProps
)(IndexPage);
