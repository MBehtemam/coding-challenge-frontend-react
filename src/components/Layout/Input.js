import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import Row from './Row';

const InputLabel = styled.label`
  background-color: #000000;
  border-radius: 10px 0px 0px 10px;
  color: #ffffff;
  padding: 12px 10px;
`;
const InputText = styled.input`
  background-clip: padding-box;
  background-color: #fff;
  border: 1px solid #ced4da;
  border-radius: 0.25rem;
  color: #495057;
  display: block;
  font-size: 1rem;
  line-height: 1.5;
  padding: 0.375rem 0rem;
  transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
  width: 90%;
`;
const Input = ({ onBlur, addOnBefore, onChange, value, defaultValue }) => (
  <Row>
    <InputLabel>{addOnBefore}</InputLabel>
    <InputText
      defaultValue={defaultValue}
      value={value}
      onBlur={e => onBlur(e.target.value)}
      onChange={e => onChange(e.target.value)}
    />
  </Row>
);

Input.propTypes = {
  onChange: PropTypes.func,
  onBlur: PropTypes.func,
  addOnBefore: PropTypes.string,
  defaultValue: PropTypes.string,
  value: PropTypes.string
};
Input.defaultProps = {
  onChange: val => val,
  onBlur: val => val,
  addOnBefore: '',
  defaultValue: '',
  value: ''
};
export default Input;
