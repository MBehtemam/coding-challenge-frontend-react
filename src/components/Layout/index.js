import styled from 'styled-components';

const layout = styled.div`
  display: flex;
  flex-direction: column;
`;

export default layout;
