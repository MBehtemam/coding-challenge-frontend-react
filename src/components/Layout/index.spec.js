import React from 'react';
import renderer from 'react-test-renderer';
import Layout from '.';

describe('Test suits for Layout component', () => {
  it('should match to snapshot', () => {
    const tree = renderer.create(<Layout />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
