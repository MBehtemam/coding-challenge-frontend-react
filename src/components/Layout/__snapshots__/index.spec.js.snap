// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suits for Layout component should match to snapshot 1`] = `
<div
  className="sc-bdVaJa bPGLpq"
/>
`;
